packages/drawer/src/ux-modal-service.ts

```diff
@@ -178,28 +202,47 @@ export class ModalService {
     const bindingContext: ModalBindingContext = {};
     const element = this.createDrawerElement(options, bindingContext);
     
-    element.innerHTML = `<compose view-model.ref=composeViewModel>`;
     if (!options.host || options.host === 'body') {
       options.host = document.body;
     } else if (typeof options.host === 'string') {
       options.host = document.querySelector(options.host) || document.body;
     }
     options.host.appendChild(element);
     let childView = this.templatingEngine.enhance({ element: element, bindingContext: bindingContext });
-    
-    if (options.viewModel && bindingContext.composeViewModel) {
-      bindingContext.composeViewModel.viewModel = options.viewModel;
-    }
-    if (options.view && bindingContext.composeViewModel) {
-      bindingContext.composeViewModel.view = options.view;
-    }
-    if (options.model && bindingContext.composeViewModel) {
-      bindingContext.composeViewModel.model = options.model;
-    }
 
+    let slot: ViewSlot;
     const controllers = (childView as any).controllers as Controller[];
-    
     const drawer: UxDrawer =  controllers[0].viewModel as UxDrawer;
+    try {
+      const view: any = controllers[0].view;
+      slot = new ViewSlot(view.slots['__au-default-slot-key__'].anchor, false);
+    } catch (_error) {
+      this.cancelOpening(drawer);
+      throw 'Missing slot in drawer';
```

Comment:
If we need to throw, let's throw a real error object instead
```suggestion
      throw new Error('Missing slot in drawer');
```
otherwise we gonna have some issues with Promise polyfills, and probably in user code too